### PR TITLE
Fix flakey spec on main

### PR DESCRIPTION
**Pull request by tomas-stefano**
## Context

We have a flakey spec: /spec/system/candidate_interface/continuous_applications/entering_details/candidate_marking_section_complete_or_incomplete_spec.rb[1:6]
‌

`with_degree` trait can create a degree of many different types. because the factory is using

```ruby
qualification_type { Hesa::DegreeType.all.sample.name }
```

This can create foundation degrees or any other degrees at random which makes the spec flakey

And on the degree wizard we don't show the complete section component if the candidate only has foundation degree for example

So forcing a bachelor or master will make the spec work

## Trello card

https://trello.com/c/NRpgI3k7/1275-investigate-and-fix-flakey-spec

### CODE REVIEW

**Comment by tomas-stefano** [comment]
[Investigate and fix flakey spec](https://trello.com/c/NRpgI3k7/1275-investigate-and-fix-flakey-spec)

File: spec/factories/application_qualification.rb [hunk 1]
```diff
@@ -116,6 +116,10 @@
         grade { Adviser::ApplicationFormValidations::APPLICABLE_DOMESTIC_DEGREE_GRADES.sample }
         qualification_level { Adviser::ApplicationFormValidations::APPLICABLE_DOMESTIC_DEGREE_LEVELS.sample }
       end
+
+      trait :bachelor do
+        qualification_type { Hesa::DegreeType.all.map(&:name).grep(/Bachelor/).first }
```
Comment (vassyz) [inline spec/factories/application_qualification.rb:121]: `Hesa::DegreeType.where(level: :bachelor).first.name?

Comment (tomas-stefano) [reply spec/factories/application_qualification.rb]: Good suggestion! Done

**Review by vassyz — APPROVED** [review]

File: spec/factories/application_qualification.rb (hunk 1 above)
Comment (tomas-stefano) [reply spec/factories/application_qualification.rb]: Funny fact: I created the where method 😂

**Pull request merged**